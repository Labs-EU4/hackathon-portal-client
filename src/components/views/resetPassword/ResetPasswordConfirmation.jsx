import React from "react";

import { BodyContainerN } from "../../../assets/styles/atoms/BodyContainerStyling";
import Container from "../../../assets/styles/atoms/ContainerStyling";
import { H1 } from "../../../assets/styles/atoms/HeadingStyling";
import { Paragraph } from "../../../assets/styles/atoms/ParagraphStyling";
import { RowBody } from "../../../assets/styles/atoms/RowBodyStyling";
import { WideBody } from "../../../assets/styles/atoms/WideBodyStyling";
import Button from "../../atoms/Button";
import HeroImage from "../../atoms/HeroImage";
import image from "../../../assets/images/Password-sent.png";

function ResetPasswordConfirmation() {
  return (
    <WideBody>
      <BodyContainer justify="center">
        <HeroImage src={image} alt="Password Sent" />
        <Container>
          <H1>Check your inbox</H1>
          <Paragraph>
            Please check your email inbox for directions on how to reset your
            password.
          </Paragraph>
          <RowBody>
            <Button anchor size="wide" to="/login">
              Return to Login page
            </Button>
          </RowBody>
        </Container>
      </BodyContainer>
    </WideBody>
  );
}

export default ResetPasswordConfirmation;