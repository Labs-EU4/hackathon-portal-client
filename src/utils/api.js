import Axios from "axios";
import { toast } from "react-toastify";
import { resetUser } from "../store/user/actions";

const baseURL = process.env.REACT_APP_API_URL;

// if process.env.CONTEXT = "production" {
//   Point to heroku API
// }
// if process.env.BRANCH = "develop" {
  // point to Heroku staging API
// }
// if process.env.CONTEXT = "deploy-preview" {
//   point to deploy-preview HEROKU API (the separate one)
// }



export const axiosWithAuth = token =>
  Axios.create({
    baseURL,
    headers: {
      Authorization: token
    }
  });

export const axios = Axios.create({ baseURL });

export const selectToken = state => state.currentUser.token;

export const showError = message => {
  toast.error(message);
};

export const showSuccess = message => {
  toast.success(message);
};

export function* handleError({ response }, put, history) {
  const { message, statusCode } = response.data;
  if (statusCode === 404) {
    history.push("/not-found");
  }
  if (message === "jwt expired") {
    yield put(resetUser());
  } else showError(`⚠️ ${message}`);
}
