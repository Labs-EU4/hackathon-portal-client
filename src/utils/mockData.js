export const initialState = {
  events: {
    data: [
      {
        id: 1,
        event_title: "EuroHack",
        event_description: "the best event hack hack hack macbook",
        creator_id: 1,
        start_date: "10/09/11",
        end_date: "08/07/13",
        location: "Europe",
        guidelines: "cool hekhkdsjfhsdkjf skdjfhskdjfhkjsdfh skjdfhksjdfh",
        participation_type: "Team",
        category_id: 1,
        requirements: [
          {
            video_url:
              "https://res.cloudinary.com/ibkrice-com/video/upload/v1581259780/Harvesting_ifato8.mp4"
          }
        ],
        ProfileImage: [
          {
            image_url:
              "https://res.cloudinary.com/ibkrice-com/video/upload/v1581259780/Harvesting_ifato8.mp4"
          }
        ],
        rubrics: [
          "Presentation",
          "Product Fit",
          "Functionality",
          "Innovation",
          "Product Design",
          "Extensibility"
        ]
      },
      {
        id: 2,
        event_title: "World",
        event_description: "number one hacker event in the world",
        creator_id: 1,
        start_date: "07/09/11",
        end_date: "08/07/13",
        location: "Asia",
        guidelines: "come well prepared,no cheating,no fighting",
        participation_type: "Team",
        category_id: 2
      }
    ],
    categories: [
      { id: 2, category_name: "Summer Hackton" },
      { id: 3, category_name: "World Hackton" },
      { id: 3, category_name: "Asia Hackton" },
      { id: 3, category_name: "Gamer's Hackton" },
      { id: 3, category_name: "Advanced Hackton" }
    ]
  },

  currentUser: {
    token: 347534574536,
    userId: 1,
    fullname: "Jake Tom",
    email: "8omemail@google.com",
    username: "Jake22",
    bio: "I like coding ",
    image: ""
  },
  submissions: [
    {
      id: 1,
      project_title: "Mock Project",
      participant_or_team_name: "Mock Team",
      event_id: 7,
      video_url: "",
      git_url: "https://www.google.com/",
      project_writeups: "No mock write up",
      submitted_by: 14,
      created_at: "2020-03-02",
      updated_at: "2020-03-02",
      average_rating: 0,
      acted_judges: 0,
      number_of_judges: 0
    }
  ],

  teams: [
    { id: 1, team_name: "The A team", team_lead: 2 },
    { id: 2, team_name: "Hacking central", team_lead: 3 },
    { id: 3, team_name: "blackhats", team_lead: 4 },
    { id: 4, team_name: "Spies", team_lead: 1 }
  ],
  participant_team_members: [
    {
      team_member: 3,
      team_id: 1
    },
    {
      team_member: 4,
      team_id: 1
    },
    {
      team_member: 5,
      team_id: 1
    }
  ]
};
