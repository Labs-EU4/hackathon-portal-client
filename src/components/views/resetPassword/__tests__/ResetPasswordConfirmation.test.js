import React from "react";
import { Router } from "react-router";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import configureStore from "redux-mock-store";
import "@testing-library/jest-dom/extend-expect";
import ResetPasswordConfirmation from "../ResetPasswordConfirmation";
import { initialState } from "../../../../utils/mockData";

const history = createMemoryHistory();

let component;
let mockStore;
let store;

beforeEach(() => {
  mockStore = configureStore();
  store = mockStore(initialState);
  component = render(
    <Router history={history}>
      <Provider store={store}>
        <ResetPasswordConfirmation />
      </Provider>
    </Router>
  );
});

describe("Asserts that text nodes on ResetPasswordConfirmation.js,render properly", () => {
  it("ResetPassword.js component renders properly", () => {
    expect(component).toMatchSnapshot();
  });
  it("The text node for the <H1> Check your inbox, renders properly ", () => {
    expect(component.queryByText(/Check your inbox/i)).toBeInTheDocument();
  });
  it("The text node for the  email confirmation, renders properly ", () => {
    expect(
      component.queryByText(
        "Please check your email inbox for directions on how to reset your password."
      )
    ).toBeInTheDocument();
  });
});
