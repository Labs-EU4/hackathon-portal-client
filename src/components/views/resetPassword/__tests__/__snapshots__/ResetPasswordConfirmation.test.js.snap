// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Asserts that text nodes on ResetPasswordConfirmation.js,render properly ResetPassword.js component renders properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-dnqmqq cbWufU"
      >
        <div
          class="sc-bdVaJa ezLspW"
        >
          <img
            alt="Password Sent"
            class="sc-VigVT goDtxb"
            src="Password-sent.png"
          />
          <div
            class="sc-htpNat hzPPtc"
          >
            <h1
              class="sc-bxivhb eytFwH"
            >
              Check your inbox
            </h1>
            <p
              class="sc-gzVnrw fREmO"
            >
              Please check your email inbox for directions on how to reset your password.
            </p>
            <div
              class="sc-htoDjs bYniKI"
            >
              <a
                class="sc-gqjmRU gRHsWV"
                to="/login"
              >
                Return to Login page
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-dnqmqq cbWufU"
    >
      <div
        class="sc-bdVaJa ezLspW"
      >
        <img
          alt="Password Sent"
          class="sc-VigVT goDtxb"
          src="Password-sent.png"
        />
        <div
          class="sc-htpNat hzPPtc"
        >
          <h1
            class="sc-bxivhb eytFwH"
          >
            Check your inbox
          </h1>
          <p
            class="sc-gzVnrw fREmO"
          >
            Please check your email inbox for directions on how to reset your password.
          </p>
          <div
            class="sc-htoDjs bYniKI"
          >
            <a
              class="sc-gqjmRU gRHsWV"
              to="/login"
            >
              Return to Login page
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
